Userland/Applets/ClipboardHistory/main.cpp

```diff
@@ -17,14 +19,23 @@
 
 ErrorOr<int> serenity_main(Main::Arguments arguments)
 {
-    TRY(Core::System::pledge("stdio recvfd sendfd rpath unix"));
+    TRY(Core::System::pledge("stdio recvfd sendfd rpath unix cpath wpath"));
     auto app = TRY(GUI::Application::try_create(arguments));
+    auto clipboard_config = TRY(Core::ConfigFile::open_for_app("ClipboardHistory"));
+
+    auto const default_path = DeprecatedString::formatted("{}/{}", Core::StandardPaths::data_directory(), "Clipboard/ClipboardHistory.json"sv);
+    auto clipboard_file_path = clipboard_config->read_entry("Clipboard", "ClipboardFilePath", default_path);
+    auto const parent_path = LexicalPath(clipboard_file_path);
+    TRY(Core::Directory::create(parent_path.dirname(), Core::Directory::CreateDirectories::Yes));
 
     Config::pledge_domain("ClipboardHistory");
     Config::monitor_domain("ClipboardHistory");
 
-    TRY(Core::System::pledge("stdio recvfd sendfd rpath"));
+    TRY(Core::System::pledge("stdio recvfd sendfd rpath cpath wpath"));
     TRY(Core::System::unveil("/res", "r"));
+    TRY(Core::System::unveil(Core::StandardPaths::config_directory(), "rwc"sv));
```

Comment:
Why are you unveiling `config_directory()` here? You read a `Core::ConfigFile` just above, before you unveil/pledge anything.

Reply:
Remains from past architecture, I removed it.
And I replaced `File::write` with `File::write_until_depleted` due to ecd18628